api/src/org/labkey/api/security/AuthenticationManager.java

```diff
@@ -125,6 +126,7 @@
     private static final Logger _log = LogManager.getLogger(AuthenticationManager.class);
     // All registered authentication providers (DbLogin, LDAP, SSO, etc.)
     private static final List<AuthenticationProvider> _allProviders = new CopyOnWriteArrayList<>();
+    private static final List<LogoutHandler> _logoutHandlers = new ArrayList<>();

```

Comment:
```suggestion
    private static final List<LogoutHandler> _logoutHandlers = new CopyOnWriteArrayList<>();

```
Using an `ArrayList` is not thread-safe

Reply:
Ended up moving this code to UserManager, where I also updated it to use a CopyOnWriteArrayList